server/container/guice/cassandra-guice/src/test/java/org/apache/james/CassandraJamesDefinition.java

```diff
@@ -0,0 +1,137 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james;
+
+import static org.apache.james.CassandraJamesServerMain.ALL_BUT_JMX_CASSANDRA_MODULE;
+
+import java.util.List;
+
+import org.apache.james.mailbox.extractor.TextExtractor;
+import org.apache.james.mailbox.store.search.PDFTextExtractor;
+import org.apache.james.modules.TestESMetricReporterModule;
+import org.apache.james.modules.TestJMAPServerModule;
+import org.apache.james.server.core.configuration.Configuration;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Module;
+import com.google.inject.util.Modules;
+
+public class CassandraJamesDefinition implements JamesServerExtension.JamesDefinition {
+
+    public static class Builder {
+        private final ImmutableList.Builder<GuiceModuleTestExtension> extensions;
+        private final ImmutableList.Builder<Module> modules;
+
+        private Builder() {
+            extensions = ImmutableList.builder();
+            modules = ImmutableList.builder();
+        }
+
+        public Builder defaultExtensions() {
```

Comment:
Just a comment: I think it's not a safe way to use the builder. When client doesn't use both `defaultExtensions()` and `addExtensions()` from Builders but there is nothing prevent that action. I mean it has to be some ways to force the client either use the default or  register extensions by him self.

Reply:
Why is it dangerous?

Why can't I use the default extensions then add the extension I need? This seems more flexible to me.

Reply:
I mean in the case the client **doesn't** use both of two methods, hence your builder is missing extensions to start james.

Reply:
answer: It is valid in some cases.